server/domains/registration.py

```diff
@@ -0,0 +1,29 @@
+from fastapi import APIRouter, Depends, HTTPException
+
+from server.database.db_settings import SessionLocal
+from server.database.db_context import DBContext
+from server.database.schemas.users import UserCreate, UserGet
+from server.database.entities.users import User
+from server import main_repo
+
+router = APIRouter()
+
+
+def db_context():
+    db = SessionLocal()
+    try:
+        yield DBContext(db)
+    finally:
+        db.close()
+
+
+@router.post("/register/", response_model=UserGet)
+def create_user(user: UserCreate, db: DBContext = Depends(db_context)):
+    db_user = main_repo.users.get_by_email(db, email=user.email)
+    if db_user:
+        raise HTTPException(status_code=400, detail="Email already registered")
+    db_user = main_repo.users.get_by_username(db, username=user.username)
```

Comment:
Is it the best practise to check if user is already registered? For me it doesn't really looks like this. Probably I'm wrong, but I think that you could create an additional repo method like `has_user`, which searches if there are users with fields, which must be unic

Reply:
Fixed